src/box/iproto.cc

```diff
@@ -557,6 +593,44 @@ struct iproto_connection
 	struct iproto_thread *iproto_thread;
 };
 
+static inline void
+errinj_stream_count_add(MAYBE_UNUSED int val)
+{
+#ifndef NDEBUG
+	struct errinj *inj =
+		errinj(ERRINJ_IPROTO_STREAM_COUNT, ERRINJ_INT);
```

Comment:
There can be several iproto threads! But there is only one STREAM_COUNT and STREAM_MSG_COUNT variable so we need atomic counters for them. Also for testing purposes, we need to have access to these variables after the stream is destroyed. Also we need to have access for them from test.

Reply:
You can maintain these counters per thread and sum them before reporting, as we do with other statistics.

Reply:
The matter of statistics reporting I would propose to postpone until someone asks for it, if you do not insist. It is significantly easier to fetch the stats via the error injection than via the public statistics and per-thread aggregation.

Reply:
Using error injection for accumulating stats in debug builds looks unnatural. AFAICS we already have all the infrastructure for accumulating and reporting iproto stats (see iproto_fill_stat). Let's just add a few more counters there for streams. They will be useful for us in release builds, because they will tell us whether a user actually enables streams and how actively they are used. They will also let us identify a memory leak if there's one caused by using streams. Let's please do this in this series, but in a separate patch with an adequate set of tests.